src/odemis/gui/cont/tabs.py

```diff
@@ -4655,6 +4653,45 @@ def terminate(self):
         for s in self.tab_data_model.streams.value:
             s.is_active.value = False
 
+    def _set_align_mode(self, evt):
+        """
+        Links the view to the model by setting the correct mode on the model whenever an alignment mode button is pressed.
+
+        :param evt (GenButtonEvent): Clicked event
+        """
+        clicked_align_button = evt.GetEventObject()
+
+        # Set the correct mode in the model
+        for mode, button in self._align_modes.items():
+            if clicked_align_button == button:
+                self.tab_data_model.align_mode.value = mode
+                return
+
+
+    def _on_align_mode(self, align_mode):
+        """
+        Subscriber for the current alignment mode. (un)toggles the correct buttons and calls the setters of each mode.
+        Which in their turn actually changes the mode with the corresponding streams to be displayed, the
+        instructions and the controls.
+
+        :param align_mode (string): Current mode (Z_ALIGN, SEM_ALIGN, FLM_ALIGN)
+        """
+        # Un/toggle all the buttons.
+        for mode, btn in self._align_modes.items():
+            if mode is not align_mode:
+                btn.SetToggle(False)
+            elif mode is align_mode:
+                btn.SetToggle(True)
+            else:
+                logging.error("Incorrect alignment mode is set.")
```

Comment:
The two tests before check every possibilities: either `mode` is `align_mode` or it's not. So this else will never happen. What you could check is that align_mode is in self._align_modes (but you'd do it outside of the loop). Alternatively, you could put an `else` in the `if` block below.

Anyway, for this loop, you can keep it simple with a code like:
```python
        for mode, btn in self._align_modes.items():
                btn.SetToggle(mode == align_mode)
```

Reply:
Neat